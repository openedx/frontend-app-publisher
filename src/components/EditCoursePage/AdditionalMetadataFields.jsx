import React from 'react';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';
import RenderInputTextField from '../RenderInputTextField';
import FieldLabel from '../FieldLabel';
import RichEditor from '../RichEditor';
import { loadOptions, utcTimeZone } from '../../utils';
import DateTimeField from '../DateTimeField';
import RenderSelectField from '../RenderSelectField';
import ReduxFormCreatableSelect from '../ReduxFormCreatableSelect';
import { courseTagValidate } from '../../utils/validation';

const AdditionalMetadataFields = (props) => {
  const {
    disabled, sourceInfo, externalCourseMarketingType, productStatusOptions, externalCourseMarketingTypeOptions,
  } = props;
  const sourceSlug = sourceInfo?.slug;

  productStatusOptions.unshift({ value: '', label: 'Select a product status' });

  externalCourseMarketingTypeOptions.unshift({ value: '', label: 'Select a marketing type' });

  const SOURCE_SLUG_REQUIRED_FIELDS = JSON.parse(process.env.ADDITIONAL_METADATA_REQUIRED_FIELDS);

  function isRequiredField(fieldName) {
    if (['certificate_info_heading', 'certificate_info_blurb'].includes(fieldName)) {
      if (externalCourseMarketingType === 'sprint') {
        return false;
      }
    }
    if (sourceSlug === undefined || !(sourceSlug in SOURCE_SLUG_REQUIRED_FIELDS)) {
      return true;
    }
    if (SOURCE_SLUG_REQUIRED_FIELDS[sourceSlug].includes(fieldName)) {
      return true;
    }
    return false;
  }

  return (
    <div className="collapsible-card pgn_collapsible mt-4">
      <div className="collapsible-body">
        <FieldLabel text="Additional Metadata Details" className="h3 font-weight-normal mb-3" />
        <div className="mb-3 mt-1">
          <Field
            name="additional_metadata.display_on_org_page"
            type="checkbox"
            label="additional_metadata.display_on_org_page"
            disabled={disabled}
            component="input"
            required={isRequiredField('display_on_org_page')}
            className="float-left mt-1 ml-1"
          />
          <FieldLabel text="Display on Organization Page" className="font-weight-normal ml-2 float-left" />
          <br />
        </div>
        <Field
          name="additional_metadata.external_url"
          component={RenderInputTextField}
          label=<FieldLabel id="external_url.label" text="External landing page URL" />
          disabled={disabled}
          required={isRequiredField('external_url')}
        />
        <Field
          name="additional_metadata.external_identifier"
          component={RenderInputTextField}
          label=<FieldLabel id="external_identifier.label" text="External system identifier" />
          disabled={disabled}
          required={isRequiredField('external_identifier')}
        />
        <Field
          name="additional_metadata.lead_capture_form_url"
          component={RenderInputTextField}
          label=<FieldLabel id="lead_capture_form_url.label" text="Lead capture form url" />
          disabled={disabled}
          required={isRequiredField('lead_capture_form_url')}
        />
        <Field
          name="additional_metadata.organic_url"
          component={RenderInputTextField}
          label=<FieldLabel id="organic_url.label" text="Organic URL" />
          disabled={disabled}
          required={isRequiredField('organic_url')}
        />
        <Field
          name="additional_metadata.variant_id"
          component={RenderInputTextField}
          label=<FieldLabel id="variant_id.label" text="Course Variant Id" />
          disabled={disabled}
          required={isRequiredField('variant_id')}
        />
        <Field
          name="additional_metadata.course_term_override"
          component={RenderInputTextField}
          label={(
            <FieldLabel
              id="course_term_override.label"
              text="Course Term Override"
              helpText={(
                <div>
                  <dl>
                    <dt>Course Term</dt>
                    <dd>
                      For example, in the UK universities use programme instead of course, so you can define which
                      term you want to use in course description to keep the consistency.
                    </dd>
                  </dl>
                </div>
              )}
            />
          )}
          disabled={disabled}
          required={isRequiredField('course_term_override')}
        />
        <FieldLabel text="Fact 1" className="h3 font-weight-normal" />
        <Field
          name="additional_metadata.facts_1_heading"
          component={RenderInputTextField}
          label=<FieldLabel id="facts.heading" text="Heading" />
          disabled={disabled}
          required={isRequiredField('facts_1_heading')}
        />
        <Field
          name="additional_metadata.facts_1_blurb"
          component={RichEditor}
          label=<FieldLabel id="facts.blurb" text="Description" />
          disabled={disabled}
          required={isRequiredField('facts_1_blurb')}
        />
        <FieldLabel text="Fact 2" className="h3 font-weight-normal" />
        <Field
          name="additional_metadata.facts_2_heading"
          component={RenderInputTextField}
          label=<FieldLabel id="facts.heading" text="Heading" />
          disabled={disabled}
          required={isRequiredField('facts_2_heading')}
        />
        <Field
          name="additional_metadata.facts_2_blurb"
          component={RichEditor}
          label=<FieldLabel id="facts.blurb" text="Description" />
          disabled={disabled}
          required={isRequiredField('facts_2_blurb')}
        />
        <FieldLabel text="Certificate Information" className="h3 font-weight-normal" />
        <Field
          name="additional_metadata.certificate_info_heading"
          component={RenderInputTextField}
          label=<FieldLabel id="certificate-info.heading" text="Heading" />
          disabled={disabled}
          required={isRequiredField('certificate_info_heading')}
        />
        <Field
          name="additional_metadata.certificate_info_blurb"
          component={RichEditor}
          label=<FieldLabel id="certificate-info.blurb" text="Description" />
          disabled={disabled}
          required={isRequiredField('certificate_info_blurb')}
        />
        <Field
          name="additional_metadata.start_date"
          type="date"
          component={DateTimeField}
          dateLabel="Start Date"
          timeLabel={`Start Time (${utcTimeZone})`}
          utcTimeZone
          disabled={disabled}
          required={isRequiredField('start_date')}
        />
        <Field
          name="additional_metadata.end_date"
          type="date"
          component={DateTimeField}
          dateLabel="End Date"
          timeLabel={`End Time (${utcTimeZone})`}
          utcTimeZone
          disabled={disabled}
          required={isRequiredField('end_date')}
        />
        <Field
          name="additional_metadata.registration_deadline"
          type="date"
          component={DateTimeField}
          dateLabel="Registration Deadline Date"
          timeLabel={`Registration Deadline Time (${utcTimeZone})`}
          utcTimeZone
          disabled={disabled}
          required={isRequiredField('registration_deadline')}
        />
        <Field
          name="additional_metadata.product_status"
          component={RenderSelectField}
          label=<FieldLabel id="product_status.label" text="Product Status" />
          options={productStatusOptions}
          value="additional_metadata.product_status"
          disabled={disabled}
          required={isRequiredField('product_status')}
        />
        <Field
          name="additional_metadata.external_course_marketing_type"
          component={RenderSelectField}
          label=<FieldLabel id="external_course_marketing_type.label" text="External Product Marketing Type" />
          options={externalCourseMarketingTypeOptions}
          value="additional_metadata.external_course_marketing_type"
          disabled={disabled}
          required={isRequiredField('external_course_marketing_type')}
        />
        <FieldLabel text="Product Meta" className="h3 font-weight-normal" />
        <Field
          name="additional_metadata.product_meta_title"
          component={RenderInputTextField}
          label=<FieldLabel id="product_meta.title" text="Title" />
          disabled={disabled}
          required={isRequiredField('product_meta')}
        />
        <Field
          name="additional_metadata.product_meta_description"
          component={RenderInputTextField}
          label=<FieldLabel id="product_meta.description" text="Description" optional />
          disabled={disabled}
          optional
        />
        <Field
          name="additional_metadata.product_meta_keywords"
          component={ReduxFormCreatableSelect}
          label={(
            <FieldLabel
              id="additional_metadata.product_meta_keywords"
              text="Keywords"
              helpText={(
                <div>
                  <dl>
                    <dt>Keywords</dt>
                    <dd>
                      Enter SEO Meta tags for Products. Separate keywords with spaces.
                    </dd>
                  </dl>
                </div>
              )}
              optional
            />
          )}
          defaultValue={[]}
          isAsync
          isMulti
          disabled={disabled}
          optional
          isCreatable
          createOptionValidator={courseTagValidate}
          loadOptions={loadOptions}
        />
      </div>
    </div>
  );
};

AdditionalMetadataFields.propTypes = {
  disabled: PropTypes.bool,
  sourceInfo: PropTypes.shape({
    name: PropTypes.string,
    slug: PropTypes.string,
    description: PropTypes.string,
  }),
  externalCourseMarketingType: PropTypes.string,
  productStatusOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
  externalCourseMarketingTypeOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
};

AdditionalMetadataFields.defaultProps = {
  disabled: false,
  sourceInfo: {},
  externalCourseMarketingType: null,
  productStatusOptions: [],
  externalCourseMarketingTypeOptions: [],
};

export default AdditionalMetadataFields;
