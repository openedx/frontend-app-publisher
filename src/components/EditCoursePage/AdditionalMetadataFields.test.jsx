import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AdditionalMetadataFields from './AdditionalMetadataFields';

const productStatusOptions = [
  { value: 'published', label: 'Published' },
  { value: 'Archived', label: 'Archived' },
];

const externalCourseMarketingTypeOptions = [
  { value: 'sprint', label: 'Sprint' },
  { value: 'short_course', label: 'Short Course' },
];

describe('AdditionalMetadata Fields', () => {
  it('Display all fields', () => {
    const component = shallow(<AdditionalMetadataFields
      productStatusOptions={productStatusOptions}
      externalCourseMarketingTypeOptions={externalCourseMarketingTypeOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('Display required fields on the basis of external course marketing type', () => {
    const component = shallow(<AdditionalMetadataFields
      externalCourseMarketingType="sprint"
      productStatusOptions={productStatusOptions}
      externalCourseMarketingTypeOptions={externalCourseMarketingTypeOptions}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
