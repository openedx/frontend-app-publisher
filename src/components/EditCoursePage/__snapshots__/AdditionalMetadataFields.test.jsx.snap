// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalMetadata Fields Display all fields 1`] = `
<div
  className="collapsible-card pgn_collapsible mt-4"
>
  <div
    className="collapsible-body"
  >
    <FieldLabel
      className="h3 font-weight-normal mb-3"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Additional Metadata Details"
    />
    <div
      className="mb-3 mt-1"
    >
      <Field
        className="float-left mt-1 ml-1"
        component="input"
        disabled={false}
        label="additional_metadata.display_on_org_page"
        name="additional_metadata.display_on_org_page"
        required={true}
        type="checkbox"
      />
      <FieldLabel
        className="font-weight-normal ml-2 float-left"
        extraText=""
        helpText=""
        id={null}
        optional={false}
        text="Display on Organization Page"
      />
      <br />
    </div>
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_url.label"
          optional={false}
          text="External landing page URL"
        />
      }
      name="additional_metadata.external_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_identifier.label"
          optional={false}
          text="External system identifier"
        />
      }
      name="additional_metadata.external_identifier"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="lead_capture_form_url.label"
          optional={false}
          text="Lead capture form url"
        />
      }
      name="additional_metadata.lead_capture_form_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="organic_url.label"
          optional={false}
          text="Organic URL"
        />
      }
      name="additional_metadata.organic_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="variant_id.label"
          optional={false}
          text="Course Variant Id"
        />
      }
      name="additional_metadata.variant_id"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <dl>
                <dt>
                  Course Term
                </dt>
                <dd>
                  For example, in the UK universities use programme instead of course, so you can define which term you want to use in course description to keep the consistency.
                </dd>
              </dl>
            </div>
          }
          id="course_term_override.label"
          optional={false}
          text="Course Term Override"
        />
      }
      name="additional_metadata.course_term_override"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Fact 1"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.facts_1_heading"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.facts_1_blurb"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Fact 2"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.facts_2_heading"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.facts_2_blurb"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Certificate Information"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="certificate-info.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.certificate_info_heading"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="certificate-info.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.certificate_info_blurb"
      required={true}
    />
    <Field
      component={[Function]}
      dateLabel="Start Date"
      disabled={false}
      name="additional_metadata.start_date"
      required={true}
      timeLabel="Start Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      dateLabel="End Date"
      disabled={false}
      name="additional_metadata.end_date"
      required={true}
      timeLabel="End Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      dateLabel="Registration Deadline Date"
      disabled={false}
      name="additional_metadata.registration_deadline"
      required={true}
      timeLabel="Registration Deadline Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_status.label"
          optional={false}
          text="Product Status"
        />
      }
      name="additional_metadata.product_status"
      options={
        Array [
          Object {
            "label": "Select a product status",
            "value": "",
          },
          Object {
            "label": "Published",
            "value": "published",
          },
          Object {
            "label": "Archived",
            "value": "Archived",
          },
        ]
      }
      required={true}
      value="additional_metadata.product_status"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_course_marketing_type.label"
          optional={false}
          text="External Product Marketing Type"
        />
      }
      name="additional_metadata.external_course_marketing_type"
      options={
        Array [
          Object {
            "label": "Select a marketing type",
            "value": "",
          },
          Object {
            "label": "Sprint",
            "value": "sprint",
          },
          Object {
            "label": "Short Course",
            "value": "short_course",
          },
        ]
      }
      required={true}
      value="additional_metadata.external_course_marketing_type"
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Product Meta"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_meta.title"
          optional={false}
          text="Title"
        />
      }
      name="additional_metadata.product_meta_title"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_meta.description"
          optional={true}
          text="Description"
        />
      }
      name="additional_metadata.product_meta_description"
      optional={true}
    />
    <Field
      component={[Function]}
      createOptionValidator={[Function]}
      defaultValue={Array []}
      disabled={false}
      isAsync={true}
      isCreatable={true}
      isMulti={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <dl>
                <dt>
                  Keywords
                </dt>
                <dd>
                  Enter SEO Meta tags for Products. Separate keywords with spaces.
                </dd>
              </dl>
            </div>
          }
          id="additional_metadata.product_meta_keywords"
          optional={true}
          text="Keywords"
        />
      }
      loadOptions={[Function]}
      name="additional_metadata.product_meta_keywords"
      optional={true}
    />
  </div>
</div>
`;

exports[`AdditionalMetadata Fields Display required fields on the basis of external course marketing type 1`] = `
<div
  className="collapsible-card pgn_collapsible mt-4"
>
  <div
    className="collapsible-body"
  >
    <FieldLabel
      className="h3 font-weight-normal mb-3"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Additional Metadata Details"
    />
    <div
      className="mb-3 mt-1"
    >
      <Field
        className="float-left mt-1 ml-1"
        component="input"
        disabled={false}
        label="additional_metadata.display_on_org_page"
        name="additional_metadata.display_on_org_page"
        required={true}
        type="checkbox"
      />
      <FieldLabel
        className="font-weight-normal ml-2 float-left"
        extraText=""
        helpText=""
        id={null}
        optional={false}
        text="Display on Organization Page"
      />
      <br />
    </div>
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_url.label"
          optional={false}
          text="External landing page URL"
        />
      }
      name="additional_metadata.external_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_identifier.label"
          optional={false}
          text="External system identifier"
        />
      }
      name="additional_metadata.external_identifier"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="lead_capture_form_url.label"
          optional={false}
          text="Lead capture form url"
        />
      }
      name="additional_metadata.lead_capture_form_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="organic_url.label"
          optional={false}
          text="Organic URL"
        />
      }
      name="additional_metadata.organic_url"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="variant_id.label"
          optional={false}
          text="Course Variant Id"
        />
      }
      name="additional_metadata.variant_id"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <dl>
                <dt>
                  Course Term
                </dt>
                <dd>
                  For example, in the UK universities use programme instead of course, so you can define which term you want to use in course description to keep the consistency.
                </dd>
              </dl>
            </div>
          }
          id="course_term_override.label"
          optional={false}
          text="Course Term Override"
        />
      }
      name="additional_metadata.course_term_override"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Fact 1"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.facts_1_heading"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.facts_1_blurb"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Fact 2"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.facts_2_heading"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="facts.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.facts_2_blurb"
      required={true}
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Certificate Information"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="certificate-info.heading"
          optional={false}
          text="Heading"
        />
      }
      name="additional_metadata.certificate_info_heading"
      required={false}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="certificate-info.blurb"
          optional={false}
          text="Description"
        />
      }
      name="additional_metadata.certificate_info_blurb"
      required={false}
    />
    <Field
      component={[Function]}
      dateLabel="Start Date"
      disabled={false}
      name="additional_metadata.start_date"
      required={true}
      timeLabel="Start Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      dateLabel="End Date"
      disabled={false}
      name="additional_metadata.end_date"
      required={true}
      timeLabel="End Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      dateLabel="Registration Deadline Date"
      disabled={false}
      name="additional_metadata.registration_deadline"
      required={true}
      timeLabel="Registration Deadline Time (UTC)"
      type="date"
      utcTimeZone={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_status.label"
          optional={false}
          text="Product Status"
        />
      }
      name="additional_metadata.product_status"
      options={
        Array [
          Object {
            "label": "Select a product status",
            "value": "",
          },
          Object {
            "label": "Select a product status",
            "value": "",
          },
          Object {
            "label": "Published",
            "value": "published",
          },
          Object {
            "label": "Archived",
            "value": "Archived",
          },
        ]
      }
      required={true}
      value="additional_metadata.product_status"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="external_course_marketing_type.label"
          optional={false}
          text="External Product Marketing Type"
        />
      }
      name="additional_metadata.external_course_marketing_type"
      options={
        Array [
          Object {
            "label": "Select a marketing type",
            "value": "",
          },
          Object {
            "label": "Select a marketing type",
            "value": "",
          },
          Object {
            "label": "Sprint",
            "value": "sprint",
          },
          Object {
            "label": "Short Course",
            "value": "short_course",
          },
        ]
      }
      required={true}
      value="additional_metadata.external_course_marketing_type"
    />
    <FieldLabel
      className="h3 font-weight-normal"
      extraText=""
      helpText=""
      id={null}
      optional={false}
      text="Product Meta"
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_meta.title"
          optional={false}
          text="Title"
        />
      }
      name="additional_metadata.product_meta_title"
      required={true}
    />
    <Field
      component={[Function]}
      disabled={false}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id="product_meta.description"
          optional={true}
          text="Description"
        />
      }
      name="additional_metadata.product_meta_description"
      optional={true}
    />
    <Field
      component={[Function]}
      createOptionValidator={[Function]}
      defaultValue={Array []}
      disabled={false}
      isAsync={true}
      isCreatable={true}
      isMulti={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <dl>
                <dt>
                  Keywords
                </dt>
                <dd>
                  Enter SEO Meta tags for Products. Separate keywords with spaces.
                </dd>
              </dl>
            </div>
          }
          id="additional_metadata.product_meta_keywords"
          optional={true}
          text="Keywords"
        />
      }
      loadOptions={[Function]}
      name="additional_metadata.product_meta_keywords"
      optional={true}
    />
  </div>
</div>
`;
